deploy/srep-vap/vcpu-overcommit/101-vcpu-overcommit-vap.yaml

```
@@ -0,0 +1,32 @@
+apiVersion: admissionregistration.k8s.io/v1
+kind: ValidatingAdmissionPolicy
+metadata:
+  name: windows-vcpu-overcommit
+spec:
+  failurePolicy: Fail
+  matchConditions:
+  - expression: (('kubevirt.io/preference-name' in object.metadata.annotations) && 
+        (object.metadata.annotations['kubevirt.io/preference-name'].lowerAscii().contains('windows'))) ||
+        (('kubevirt.io/cluster-preference-name' in object.metadata.annotations) && 
+        (object.metadata.annotations['kubevirt.io/cluster-preference-name'].lowerAscii().contains('windows'))) ||
+        (('vm.kubevirt.io/os' in object.metadata.annotations) && 
+        (object.metadata.annotations['vm.kubevirt.io/os'].lowerAscii().contains('windows')))
+    name: windows-vcpu-overcommit
+  matchConstraints:
+    resourceRules:
+    - apiGroups: ["kubevirt.io"]
+      apiVersions: ["*"]
+      operations: ["CREATE", "UPDATE"]
+      resources: ["virtualmachineinstances"]
+  validations:
+    - expression: |-
+        (
+          'kubevirt.io/cluster-preference-name' in object.metadata.annotations &&
+          object.metadata.annotations['kubevirt.io/cluster-preference-name'].lowerAscii().contains('dedicated')
+        ) ||
+        (
+          'kubevirt.io/preference-name' in object.metadata.annotations &&
+          object.metadata.annotations['kubevirt.io/preference-name'].lowerAscii().contains('dedicated')
+        )
+      message: "Windows VM are required to use *dedicated preferences."
```

```suggestion    
    - expression: |-
        (
          'kubevirt.io/cluster-preference-name' in object.metadata.annotations &&
          object.metadata.annotations['kubevirt.io/cluster-preference-name'].lowerAscii().contains('dedicated')
        ) ||
        (
          'kubevirt.io/preference-name' in object.metadata.annotations &&
          object.metadata.annotations['kubevirt.io/preference-name'].lowerAscii().contains('dedicated')
        )
      message: "Windows VM are required to use *dedicated preferences."
    - expresion: |- 
        has(object.spec.domain.cpu.dedicatedCpuPlacement) && object.spec.domain.cpu.dedicatedCpuPlacement == true
        message: "Windows VMIs require dedicated CPU placement. Set spec.domain.cpu.dedicatedCpuPlacement to true."
```

We can add another further validation expression. I didn't think about the fact that users may override preferences if they specify a different `dedicatedCpuPlacement` configuration in the spec of the VM. Sorry about that, this is my bad.